### PR TITLE
Add nginx-module-vts input plugin.

### PR DESCRIPTION
**Pull request by monder**
Plugin structure based on `nginx_plus` input plugin.
Nginx module source and documentation: https://github.com/vozlt/nginx-module-vts

### Required for all PRs:

- [X] Signed [CLA](https://influxdata.com/community/cla/).
- [X] Associated README.md updated.
- [X] Has appropriate unit tests.

### CODE REVIEW

**Comment by szibis** [comment]
When this #PR have a chance to be merged? in next 1.8.0 maybe?

File: plugins/inputs/nginx_vts/README.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+```
+
+When run with:
+```
+./telegraf -config telegraf.conf -input-filter nginx_vts -test
+```
+
+It produces:
+```
+* Plugin: inputs.nginx_vts, Collection 1
```
Comment (glinton) [inline plugins/inputs/nginx_vts/README.md]: For example output, we are trying to strip out the command used, the plugin prefix (`Plugin: inputs.thing`) and the leading `> `. Can you remove that from this output please.

File: plugins/inputs/nginx_vts/nginx_vts.go [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+	client *http.Client
+
+	ResponseTimeout internal.Duration
+}
+
+var sampleConfig = `
+  ## An array of ngx_http_status_module or status URI to gather stats.
+  urls = ["http://localhost/status"]
+
+  # HTTP response timeout (default: 5s)
```
Comment (glinton) [inline plugins/inputs/nginx_vts/nginx_vts.go]: Use two pounds for config comments - (`##`) and we generally comment default settings out in the sampleconfig if you'd like to do that too.

File: plugins/inputs/nginx_vts/nginx_vts.go [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+			acc.AddError(n.gatherURL(addr, acc))
+		}(addr)
+	}
+
+	wg.Wait()
+	return nil
+}
+
+func (n *NginxVTS) createHTTPClient() (*http.Client, error) {
+
```
Comment (glinton) [inline plugins/inputs/nginx_vts/nginx_vts.go]: Remove this empty line

File: plugins/inputs/nginx_vts/nginx_vts.go [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+		Transport: &http.Transport{},
+		Timeout:   n.ResponseTimeout.Duration,
+	}
+
+	return client, nil
+}
+
+func (n *NginxVTS) gatherURL(addr *url.URL, acc telegraf.Accumulator) error {
+	resp, err := n.client.Get(addr.String())
+
```
Comment (glinton) [inline plugins/inputs/nginx_vts/nginx_vts.go]: Move this newline to under your error check block.

File: plugins/inputs/nginx_vts/nginx_vts.go [hunk 5]
```diff
@@ -0,0 +1,344 @@
...
+	contentType := strings.Split(resp.Header.Get("Content-Type"), ";")[0]
+	switch contentType {
+	case "application/json":
+		return gatherStatusURL(bufio.NewReader(resp.Body), getTags(addr), acc)
+	default:
+		return fmt.Errorf("%s returned unexpected content type %s", addr.String(), contentType)
+	}
+}
+
+type NginxVts struct {
```
Comment (glinton) [inline plugins/inputs/nginx_vts/nginx_vts.go]: Use a different name for this or the other, the case of two letters is too close to differentiate.

File: plugins/inputs/nginx_vts/nginx_vts.go [hunk 6]
```diff
@@ -0,0 +1,344 @@
...
+	dec := json.NewDecoder(r)
+	status := &NginxVts{}
+	if err := dec.Decode(status); err != nil {
+		return fmt.Errorf("Error while decoding JSON response")
+	}
+	status.Gather(tags, acc)
+	return nil
+}
+
+func (s *NginxVts) Gather(tags map[string]string, acc telegraf.Accumulator) {
```
Comment (glinton) [inline plugins/inputs/nginx_vts/nginx_vts.go]: Maybe just make this part of the `gatherStatusURL` function.

**Comment by szibis** [comment]
Any updates ?

**Comment by glinton** [comment]
I either spaced it or didn't have time to re-review the entire pr (due to the force push), I'll try getting to this today

**Comment by szibis** [comment]
Is it possible to have this in upcoming 1.9.0 ??

**Comment by glinton** [comment]
Yes, it's very likely it will be in 1.9.0

File: plugins/inputs/nginx_vts/README.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+  - in_bytes
+  - out_bytes
+  - response_1xx_count
+  - response_2xx_count
+  - response_3xx_count
+  - response_4xx_count
+  - response_5xx_count
+  - weight
+  - maxFails
+  - failTimeout
```
Comment (danielnelson) [inline plugins/inputs/nginx_vts/README.md]: Convert any `camelCase` fields into `snake_case`.

**Comment by szibis** [comment]
@monder can you look ??

**Pull request merged**